doc/starlight: add RIOT changelog

LasseRosenow opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md.
-->

### Contribution description

Added the changelog of riot releases to the guide page and deprecated the doxygen page

Nice cool new fetures:
- Changelog is rendered and not in pure markdown anymore
- Changelog page has a list of all releases and you can click to see the changelog of a single release
- WE HAVE AN RSS FEED NOW!!!!!!111!!!111!!111
- I also enabled emojis for the markdown parsing, since I was already playing around with markdown rendering and stuff

### Screenshots
<img width="2525" height="1322" alt="image" src="https://github.com/user-attachments/assets/4ccc7a86-5650-481e-8cbb-949cf84c9ece" />

<img width="2525" height="1322" alt="image" src="https://github.com/user-attachments/assets/f2e74cff-15af-4996-8742-565316f5e6a8" />

LasseRosenow commented on doc/starlight/src/pages/misc/changelog.astro:
```diff
@@ -0,0 +1,32 @@
+  >
+    Subscribe to RSS Feed
+  </LinkButton>
+  <br /><br />
```
I only measured a few pixel difference, but anyways I now fixed it and I can confirm, that the margin between the button and the content border is now equal to the margin of the button and the next entry:
<img width="512" height="451" alt="image" src="https://github.com/user-attachments/assets/b7ff4847-c94b-4855-bb72-8c1c080f2920" />

LasseRosenow replied:
Done

LasseRosenow commented on doc/starlight/src/pages/misc/changelog/rss.xml.js:
```diff
@@ -0,0 +1,24 @@
+      link: `/misc/changelog/${entry.data.slug}`,
+    })),
+    // (optional) inject custom xml
+    customData: `<language>en-us</language>`,
```
Done

LasseRosenow commented on doc/starlight/src/content.config.ts:
```diff
@@ -1,10 +1,71 @@
+              .use(remarkParse)
+              .use(remarkRehype, { allowDangerousHtml: true })
+              .use(rehypeStringify)
+              .processSync(markdown);
```
Done

AnnsAnns commented on doc/starlight/src/pages/misc/changelog/[version].astro:
```diff
@@ -0,0 +1,27 @@
+  frontmatter={{
+    title: entry.data.title,
+  }}
+  ><Content />
```
```suggestion
  >
    <Content />
```
Is this the default formatting? Feels weird to have a > directly followed by the Content, Id have expected that to be a this.

AnnsAnns commented on doc/doxygen/src/changelog-deprecated.md:
```diff
@@ -0,0 +1,6 @@
+@deprecated This page is deprecated.
+See [Changelog](https://guide.riot-os.org/misc/changelog)
+on the RIOT Guide Site for the latest information.
+This page will be removed after release 2026.01.
```
```suggestion
This page will be removed after release 2026.04.
```

If you want to conform with the other deprecations that should be 2026.04, (2 releases in the future)

LasseRosenow replied:
Fixed

AnnsAnns commented on doc/starlight/src/pages/misc/changelog.astro:
```diff
@@ -0,0 +1,32 @@
+  }}
+>
+  <LinkButton
+    href="/misc/changelog/rss.xml"
```
We talked about this *in the real world* but this might look better if Changelog is a root dir entry below the Misc section, esp. since its a custom page as is already

LasseRosenow replied:
Done

LasseRosenow commented on doc/starlight/src/pages/changelog/[version].astro:
```diff
@@ -0,0 +1,27 @@
+
+<StarlightPage
+  frontmatter={{
+    title: entry.data.title,
```
Okay I added a description. Not sure really what to write there though, so maybe you don't like it :D

LasseRosenow commented on doc/starlight/src/pages/changelog/rss.xml.js:
```diff
@@ -0,0 +1,24 @@
+    site: context.site,
+    // Array of `<item>`s in output xml
+    // See "Generating items" section for examples using content collections and glob imports
+    items: changelog.map((entry) => ({
```
Okay I added a date to the rss feed

LasseRosenow commented on doc/starlight/src/pages/changelog.astro:
```diff
@@ -0,0 +1,32 @@
+  >
+    Subscribe to RSS Feed
+  </LinkButton>
+  <div style={{ height: "0.5rem" }}></div>
```
Done

LasseRosenow commented:
Ready for review :)

LasseRosenow commented:
Okay a small fix to the comments, but now it is ready

crasbe commented on doc/starlight/src/content.config.ts:
```diff
@@ -29,86 +37,82 @@ export function changelogLoader(): Loader {
+        // Convert all headings to be one level lower
+        // h1 => h2, h2 => h3, h3 => h4
+        // This is necessary, because the heading levels of this file are wrong (for markdown)
+        // We cannot have more han one h1 per document etc.
```
```suggestion
        // We cannot have more than one h1 per document etc.
```

LasseRosenow replied:
Fixed :)

crasbe commented on doc/starlight/src/content.config.ts:
```diff
@@ -29,86 +37,82 @@ export function changelogLoader(): Loader {
+          /(RIOT-.*? - Release Notes)|(Release 2013\.08)/.test(line) ||
+          index == lines.length - 1
+        ) {
+          // If we already found a release heading before we now
```
```suggestion
          // If we already found a release heading before, we now
```

LasseRosenow replied:
Fixed

AnnsAnns reviewed (approved):
Lasse asked me to quickly check again after the rewrite of the way he handles the parser but there is nothing that I can really complain about anymore that didn't feel like extreme nitpicking so uh, hope this gets merged thank you for this :frog: :tada:

LasseRosenow commented:
> (There is one commit called Fix Changes. You could also rename it.)

Yeah my git created some problems. I have fixed it now. :)

Squashed squashed squassshhhhed :)
